esp-hal/build.rs

```diff
@@ -77,6 +77,28 @@ fn main() -> Result<(), Box<dyn Error>> {
             Value::Bool(false),
             None
         ),
+        // Ideally, we should be able to set any clock frequency for any chip. However, currently
+        // only the 32 and C2 implements any sort of configurability, and the rest have a fixed
+        // clock frequeny.
+        // TODO: only show this configuration for chips that have multiple valid options.
+        (
+            "xtal-frequency",
+            "The frequency of the crystal oscillator, in MHz. Set to `auto` to automatically detect the frequency. `auto` may not be able to identify the clock frequency in some cases. Also, configuring a specific frequency may increase performance slightly.",
+            Value::String(match device_name {
```

Comment:
Are we going to run into issues if someone wants to set this config but also support multiple different chips?

Reply:
We won't, but they will

Reply:
I guess #3020 will support this scenario better